src/main/java/ch/uzh/ifi/hase/soprafs21/service/SchedulingService.java

```diff
@@ -0,0 +1,182 @@
+package ch.uzh.ifi.hase.soprafs21.service;
+
+import ch.uzh.ifi.hase.soprafs21.constant.SwipeStatus;
+import ch.uzh.ifi.hase.soprafs21.entities.*;
+import ch.uzh.ifi.hase.soprafs21.repository.ActivityRepository;
+import ch.uzh.ifi.hase.soprafs21.repository.ScheduledActivityRepository;
+import ch.uzh.ifi.hase.soprafs21.repository.SchedulingSessionRepository;
+import ch.uzh.ifi.hase.soprafs21.repository.UserRepository;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.server.ResponseStatusException;
+
+import javax.validation.ConstraintViolationException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * Scheduling Service.
+ */
+@Service
+@Transactional
+public class SchedulingService {
+
+    private final Logger log = LoggerFactory.getLogger(SchedulingService.class);
+
+    private final SchedulingSessionRepository schedulingSessionRepository;
+
+    private final ScheduledActivityRepository scheduledActivityRepository;
+
+    private final ActivityRepository activityRepository;
+
+    private final UserRepository userRepository;
+
+    @Autowired
+    public SchedulingService(@Qualifier("schedulingSessionRepository") SchedulingSessionRepository schedulingSessionRepository,
+                             @Qualifier("scheduledActivityRepository") ScheduledActivityRepository scheduledActivityRepository,
+                             @Qualifier("activityRepository") ActivityRepository activityRepository,
+                             @Qualifier("userRepository") UserRepository userRepository) {
+        this.schedulingSessionRepository = schedulingSessionRepository;
+        this.scheduledActivityRepository = scheduledActivityRepository;
+        this.activityRepository = activityRepository;
+        this.userRepository = userRepository;
+    }
+
+    public SchedulingSession createSchedulingSession(Long userId1, Long userId2, String token) {
+        SchedulingSession schedulingSession = new SchedulingSession();
+        User user = userRepository.findByToken(token);
+        if (!user.getId().equals(userId1) && !user.getId().equals(userId2)) {
+            throw new ResponseStatusException(HttpStatus.UNAUTHORIZED, "User can not start scheduling Session of other users.");
+        }
+
+        List<Activity> activityList = activityRepository.findAll();
```

Comment:
Here you could use my getAllActivitiesWithMatchedUsers method that i have implemented in the second pull request in the ActivityService.
My method would return all the activities of a given user where both users in the activity swiped TRUE. Using my method would give a smaller activityList and the for loop below would have to iterate over less activites and therefore  the looping may be faster (instead if you take all the activities like here using findAll(), the for loop below has to iterate over all the acitvities we have, which is not necessary since we are not interested in all the activities)

Reply:
Will do.